### DISCUSSION TITLE
IS there any way to map the xls column with name rather than index

### DISCUSSION THREAD

**Discussion by BBLearnerPASM** [sskorol/test-data-supplier#118] [Q&A] [answered]
IS there any way to map the xls column with COLUMN NAME rather than index?

The reason for asking is if I need to add the column randomly then I have redo the index in the java file , its tedious if I have more columns .

Any idea will be appraciated .

thanks 
BB

**Comment by sskorol** [comment]
According to ZeroCell [column code](https://github.com/creditdatamw/zerocell/blob/90a33370069a29054058091be26a5f3e0eb22814/core/src/main/java/com/creditdatamw/zerocell/annotation/Column.java#L27), index is mandatory and used in several other places for mapping. But I agree with you that it's not really flexible. So you may want to file a feature request within ZeroCell repo. test-data-supplier just wraps it, so I can't do anything on my side unfortunately. Theoretically, it might be possible to hack things by reinventing the wheel. But it'll increase the maintenance cost.

**Comment by sskorol** [comment, marked as answer]
@BBLearnerPASM I've recently release a new 2.1.0 version with custom excel processor implementation w/o ZeroCell. So now it's possible to map columns by name only, w/o indexes. See details in [docs](https://github.com/sskorol/test-data-supplier#json-csv-yaml-and-xlsx-processors).

**Comment by BBLearnerPASM** [comment]
Wow !! hats off mate !!
Will give a try this weekend and come back to you with the feedback .

It's really great achievement . As a user I am happy .

Regards,
BB
